executor/insert_common.go

```diff
@@ -562,7 +561,13 @@ func (e *InsertValues) batchCheckAndInsert(rows [][]types.Datum, addRecord func(
 		// There may be duplicate keys inside the insert statement.
 		if rows[i] != nil {
 			e.ctx.GetSessionVars().StmtCtx.AddCopiedRows(1)
-			_, err = addRecord(rows[i])
+			// All the rows are checked, so it is safe to skip the constraint check when adding record.
+			_, err = addRecord(rows[i], &table.AddRecordOpt{
+				CreateIdxOpt: table.CreateIdxOpt{
+					SkipHandleCheck: true,
+					SkipCheck:       true,
```

Comment:
maybe `SkipUniqueIndexCheck` is better :D at last if we have many skip flag maybe better use one byte xor, but current way is good to me too